.github/workflows/regression-testing/regressions-util.py

```diff
@@ -0,0 +1,494 @@
+     'metric', 'mode', 'version', 'params', 'app_version', 'utc', 'duration',
+     'training_duration', 'predict_duration', 'models_count', 'seed', 'info',
+     'acc', 'auc', 'balacc', 'logloss', 'mae', 'r2', 'rmse']
+
+
+    Parameters
+    ----------
+    baseline_csv_classification: path
+        Path to the csv containing the baseline classification results
+
+    baseline_csv_regression: path
+        Path to the csv containing the baseline regression results
+
+    targeted_csv_classification: path
+        Path to the csv containing the targeted classification results
+
+    targeted_csv_regression: path
+        Path to the csv containing the targeted regression results
+
+    Returns
+    -------
+    Tuple[df.DataFrame, df.DataFrame, df.DataFrame]
+        Returns the mean results across folds for:
+
+            baseline, targeted, comparisons
+
+        Comparisons here is the difference between (targeted - baseline)
+        Returns them in that specific order
+    """
+
+
+    # Load in data and get the means across folds
+    df_baseline_classificiation = pd.read_csv(baseline_csv_classification)
+    df_baseline_regression = pd.read_csv(baseline_csv_regression)
+    df_baseline = pd.concat([df_baseline_classificiation, df_baseline_regression])
+
+    df_baseline_means = _get_mean_results_across_folds(df_baseline)
+
+    df_targeted_classificiation = pd.read_csv(targeted_csv_classification)
+    df_targeted_regression = pd.read_csv(targeted_csv_regression)
+    df_targeted = pd.concat([df_targeted_classificiation, df_targeted_regression])
+    df_targeted_means = _get_mean_results_across_folds(df_targeted)
+
+    # Find the set intersection of tasks they have in common
+    common_tasks = set(df_baseline_means.index).intersection(set(df_targeted_means.index))
+    missing_tasks = set(df_baseline_means.index) - common_tasks
+
+    # Find the set of metrics that are comparable
+    baseline_metrics = set(METRICS).intersection(set(df_baseline_means.columns))
+    common_metrics = baseline_metrics.intersection(set(df_targeted_means.columns))
+    missing_metrics = baseline_metrics - set(df_targeted_means.columns)
+
+    # Calculate the differences for in common tasks, across all available metrics
+    df_differences = df_targeted_means.loc[common_tasks][common_metrics] \
+        - df_baseline_means.loc[common_tasks][common_metrics]
+
+    # Get the metric used for training and the dataset id of common tasks
+    df_info = df_baseline_means.loc[common_tasks][['id', 'metric']]
+
+    # Join together the info and the differences
+    return df_baseline_means, df_targeted_means, df_info.join(df_differences)
+
+def create_comparisons_markdown(
+    baseline_means_csv: str,
+    targeted_means_csv: str,
+    compared_means_csv: str,
+) -> str:
+    """ Creates markdown that can be posted to Github that shows
+    a comparison between the baseline and the targeted branch.
+
+    Parameters
+    ----------
+    baseline_means_csv: path
```

Comment:
Above it's a `str` and here it's a `path`, shouldn't you simply convert to `pathlib.Path` before passing this in here?

Reply:
I changed the docstring to use `str`, we could go with Path objects but we wouldn't use any of it's functionality